src/Chess.hpp

```diff
@@ -0,0 +1,134 @@
+#include <vector>
+
+#ifndef CHESS_VR_CHESS_HPP
+#define CHESS_VR_CHESS_HPP
+
+namespace aic
+{
+class Piece {
+
+public:
+    enum class Type {
+        KING,
+        QUEEN,
+        ROOK,
+        BISHOP,
+        KNIGHT,
+        PAWN
+    };
+    enum class Color {
+        WHITE,
+        BLACK
+    };
+
+    explicit Piece(Type t, Color col, char colum, int lvl);
+
+    /*
+     * Set the piece Type.
+     *
+     * @param t Piece Type (KING, QUEEN ...).
+     * @return void.
+     */
+    void setPieceType(Type t);
+
+    /*
+     * Set the piece Color.
+     *
+     * @param col Piece Color (WHITE | BLACK).
+     * @return void.
+     */
+    void setPieceColor(Color col);
+
+    /*
+     * Set the piece column location.
+     *
+     * @param colum char letter (A-H).
+     * @return void.
+     */
+    void setPieceColumn(char colum);
+
+    /*
+     * Set the piece level location.
+     *
+     * @param lvl Level (row) as int (1-8).
+     * @return void.
+     */
+    void setPieceLevel(int lvl);
+
+    Piece::Type getPieceType()
+    {
+        return this->type;
+    }
+
+    Piece::Color getPieceColor()
+    {
+        return this->color;
+    }
+
+    char getPieceCol()
+    {
+        return this->column;
+    }
+
+    int getPieceLvl()
+    {
+        return this->level;
+    }
+
+private:
+    Piece::Type type;
+    Piece::Color color;
+    char column;
```

Comment:
Lets switch to chess official terminology - columns are files and rows are ranks

Reply:
Renamed.